caddytest/caddytest.go

```diff
@@ -19,6 +19,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/aryann/difflib"
```

Comment:
How necessary is this dependency?

Reply:
It makes understanding what is broken significantly easier. I'm not sold on this particular tool, could use a json diffing tool which should not require the sorting code.

Reply:
Should/could the sorting be done in the tests instead of in the actual code? Not saying it's bad in the code, just wondering since it's extra work just to cater to the tests.

Reply:
I would view it as reducing the friction to writing tests. One reason to exclude it would be because it could end up in the binary. Not sure if the compiler understands that this for tests only.
The alternative would be to include a function in a test file that does the diffing instead. Not sure if that would be materially different.